src/aead/aes_gcm.rs

```diff
@@ -167,29 +167,26 @@ fn aead(
         let mut input = Block::zero();
         input.overwrite_part_at(0, remainder);
         if let Direction::Opening { .. } = direction {
-            gcm_ctx.update_block(input);
+            gcm_ctx.update_block(gcm::bytes_to_u64s(*input.as_ref()));
         }
         let mut output = aes_key.encrypt_iv_xor_block(ctr.into(), input);
         if let Direction::Sealing = direction {
             output.zero_from(remainder.len());
-            gcm_ctx.update_block(output);
+            gcm_ctx.update_block(gcm::bytes_to_u64s(*output.as_ref()));
         }
         output
     });
 
     // Authenticate the final block containing the input lengths.
     let aad_bits = polyfill::u64_from_usize(aad_len) << 3;
     let ciphertext_bits = polyfill::u64_from_usize(total_in_out_len) << 3;
-    gcm_ctx.update_block(Block::from_u64_be(
-        BigEndian::from(aad_bits),
-        BigEndian::from(ciphertext_bits),
-    ));
+    gcm_ctx.update_block([aad_bits.into(), ciphertext_bits.into()]);
 
     // Finalize the tag and return it.
     gcm_ctx.pre_finish(|pre_tag| {
         let bytes = tag_iv.into_bytes_less_safe();
         let mut tag = aes_key.encrypt_block(Block::from(&bytes));
```

Comment:
WDYT about using `aes_key.encrypt_block((&bytes).into())` here?

Reply:
Since the end goal is to remove `Block`, I think using `Block::from` here is better, because that helps us see what we need to change. If we use `Into::into()` instead then the code maybe would still compile even after we remove `Block` even if we don't remove the `.into()`.